src/validateIAPToken.ts

```diff
@@ -46,12 +46,22 @@ export async function validateIAPToken(
     throw Error(`Invalid IAP issuer [${decoded.payload.iss}]`);
   }
 
+  const expectedAudienceArray = Array.isArray(expectedAudiences)
+    ? expectedAudiences
+    : [expectedAudiences];
+  const aud = decoded.payload.aud;
+
   // From: https://cloud.google.com/iap/docs/signed-headers-howto
   // `aud` Must be a string with the following values:
   //  App Engine: /projects/PROJECT_NUMBER/apps/PROJECT_ID
   //  Compute Engine and GKE: /projects/PROJECT_NUMBER/global/backendServices/SERVICE_ID
-  if (decoded.payload.aud !== expectedAudience) {
-    throw Error(`Invalid audience [${decoded.payload.aud}]`);
+  if (
```

Comment:
All of this might be simplified to:
```suggestion
  if (!expectedAudiences.includes(decoded.payload.aud))
```

From the comment above, it seems the decoded `aud` should be a string containing only one audience for the given JWT token.

Reply:
I'm getting
```
TS2345: Argument of type  string | string[] | undefined  is not assignable to parameter of type  string 
Type  undefined  is not assignable to type  string
```

Reply:
From reading the link that's mentioned in the comment, it seems it's expected from us to validate that the decoded audience is indeed a string.

So what you could do is:
```suggestion
  if (typeof decoded.payload.aud !== "string") {
    throw Error(`Invalid audience: expected a string, received [${decoded.payload.aud}]`);
 }
 if (!expectedAudiences.includes(decoded.payload.aud)) {
 ...
```

Reply:
Based on https://datatracker.ietf.org/doc/html/rfc7519#section-4.1.3
```
 In the general case, the "aud" value is an array of case-
   sensitive strings, each containing a StringOrURI value.  In the
   special case when the JWT has one audience
```
So I think we should allow array

Reply:
But we are validating only IAP tokens with this lib, right? Or you plan to use it for something else than IAP?

https://cloud.google.com/iap/docs/signed-headers-howto#verifying_the_jwt_payload
<img width="872" alt="Screenshot 2023-09-21 at 16 04 35" src="https://github.com/kiwicom/js-iam-middleware/assets/22741774/800bdf92-f4ca-4001-a3c1-50f18ebf2bd9">

Reply:
No. My use case was just to be ready for migration to platform API, where the app would probably need to run for some time in 'dual mode' to allow multiple audiences from IAP.

I thought this was a more generic library and I don't know if it's used only in Kiwi or by someone else. But when I look on readme, I see it's probably just kiwi related, then I don't know why it's public, but that's out of scope of this MR :)

So it's up to you. If you want to allow only single audience from IAP, I'll change it

Reply:
> I thought this was a more generic library and I don't know if it's used only in Kiwi or by someone else. But when I look on readme, I see it's probably just kiwi related, then I don't know why it's public, but that's out of scope of this MR :)

Well, there are many things to say about this :)

Given the comment in code and the previous behaviour, I would not support multiple audiences decoded from the JWT token and only the middleware's functionality to allow multiple audiences, i.e. validate that the JWT tokens have an audience taken from a provided allow-list.